jujugui/static/gui/src/app/components/login/login.js

```diff
@@ -23,14 +23,28 @@ YUI.add('login-component', function() {
   juju.components.Login = React.createClass({
 
     propTypes: {
+      controllerAPI: React.PropTypes.object.isRequired,
       errorMessage: React.PropTypes.string,
+      gisf: React.PropTypes.bool.isRequired,
       isLegacyJuju: React.PropTypes.bool.isRequired,
+      localStorageGet: React.PropTypes.func,
       loginToAPIs: React.PropTypes.func.isRequired,
-      loginToController: React.PropTypes.func.isRequired
+      loginToController: React.PropTypes.func.isRequired,
+      sendPost: React.PropTypes.func
     },
 
     componentDidMount: function () {
       this.refs.username.focus();
```

Comment:
We can't actually return here; there's still the callback bit which should execute whether we're in gisf or not, but needs to happen after this postback if it exists.